test/select.jl

```diff
@@ -619,12 +619,12 @@ end
 
     df = DataFrame(rand(10, 4))
     df2 = select(df, :, :x1 => :x3)
-    @test df2 == DataFrame(eachcol(df)[[1,2,1,4]])
+    @test df2 == DataFrame(collect(eachcol(df))[[1,2,1,4]])
```

Comment:
What does this do without `collect`?

Reply:
`eachcol(df)[[1,2,1,4]]` will error, because multi column index on `eachcol(df)` will create a new data frame and return a `DataFrameColumns` wrapper of it, and in this case we have a duplicate index `1`, which will error when creating an internal data frame.

Reply:
@nalimilan - have a look at last commit. Making `eachcol(df)[columns]` return a `DataFrameColumns` object based on a view of the parent is quite appealing, but it also has the consequence that the type of column when you iterate it changes (because we have a `SubDataFrame` underneath so its columns are views, not original columns).

I still like the view approach, as it is more "correct" in terms of design. However, it has a drawback of changing the type of the vector (and probably also some small performance penalty). The alternative would be to make a copy of the parent without copying the columns. Then all is fast and column types are preserved, but the cost is that we "detach" this new data frame from the source (which we typically try to avoid - but maybe in this case we can accept it?).

Reply:
That's an interesting design issue. :-)

I see the point of creating a `SubDataFrame` parent, but it's weird that the resulting `DataFrameColumns` object has `SubArray` elements: getting different types for `x[1]` and `y[1]` when doing e.g. `x = eachcol(df); y = x[1:2]` isn't ideal. Also, if you want to access the original parent, you have to do `parent(parent(y))`, so anyway you can't use the same code for `x` and `y`. Can you think of scenarios where this would be useful?